packages/grafana-data/src/panel/PanelPlugin.ts

```diff
@@ -170,22 +224,37 @@ export class PanelPlugin<TOptions = any> extends GrafanaPlugin<PanelPluginMeta>
    *
    * @public
    **/
-  setPanelOptions(builder: (builder: PanelOptionsEditorBuilder) => void) {
+  setPanelOptions(builder: (builder: PanelOptionsEditorBuilder<TOptions>) => void) {
     // builder is applied lazily when options UI is created
     this.registerOptionEditors = builder;
     return this;
   }
 
   /**
-   * Enables configuration of panel's default field config
+   * Allows specyfing which common field config options panel should use
+   *
+   * @example
+   * ```typescript
+   *
+   * import { ShapePanel } from './ShapePanel';
+   *
+   * interface ShapePanelOptions {}
+   *
+   * // when plugin should use only specific common options
```

Comment:
```suggestion
   * // when plugin should only display specific common options
```

IIUC, all the common fields are *used*, but this just controls what shows up in the UI

Reply:
Yes, you are right